scripts/saml-configuration

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import logging
+import os
+import pwd
+import sys
+from typing import Any, Dict, Iterator, List, Tuple
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(BASE_DIR)
+# Enter the Zulip virtual environment.
+from scripts.lib.setup_path import setup_path
+
+setup_path()
+
+# Settings for django.conf
+os.environ['DJANGO_SETTINGS_MODULE'] = 'zproject.settings'
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from django.conf import settings
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser as Parser
+from onelogin.saml2.utils import OneLogin_Saml2_Utils as Utils
+
+"""A script to manage some of the SAML configuration automatically.
+
+Current status:
+Keep the signing certificates of the IdPs listed in
+settings.SOCIAL_AUTH_SAML_ENABLED_IDPS up to date.
+"""
+
+
+# The template for the path to the IdP certificate.
+idp_cert_path_template: str = '/etc/zulip/saml/idps/{}.crt'
+# The format string used for logging.
+log_format: str = '%(levelname)s:%(asctime)s saml-configuration: %(message)s'
+
+
+def is_valid(cert: x509.Certificate) -> bool:
+    '''Check if the given certificate is currently valid.'''
+    now: datetime.datetime = datetime.datetime.now()
+    return now >= cert.not_valid_before and now <= cert.not_valid_after
+
+
+parser: argparse.ArgumentParser = argparse.ArgumentParser()
+parser.add_argument(
+    '--debug', action = 'store_true', help = 'enable debug logging'
+)
+args: argparse.Namespace = parser.parse_args()
+
+if args.debug:
+    logging.basicConfig(format = log_format, level = logging.DEBUG)
+else:
+    logging.basicConfig(format = log_format)
+
+if pwd.getpwuid(os.getuid()).pw_name != 'zulip':
+    logging.critical('This script needs to be executed as user \'zulip\'')
+    sys.exit(1)
+
+# The dictionary storing the enabled IdPs (using their names as keys).
+try:
+    idps: Dict[str, Any] = settings.SOCIAL_AUTH_SAML_ENABLED_IDPS
+except Exception:
```

Comment:
I don't think we need this - SOCIAL_AUTH_SAML_ENABLED_IDPS should always be accessible - and if it isn't, we can let the exception happen normally without catching it

Reply:
Ok, I will change this!